mlir/include/mlir/IR/OpDefinition.h

```diff
@@ -1268,6 +1269,16 @@ class AffineScope : public TraitBase<ConcreteType, AffineScope> {
   }
 };
 
+/// A trait of operation. Any operation holds the AffineSymbol, and its result
+/// can be used as a symbol.
```

Comment:
This will not be sufficient documentation. Expand it with full context including the lines from your comment on the expected properties of the operation. For e.g., it does not make sense to mark something that could take an operand (and potentially an affine.for IV operand from the scope) with the AffineSymbol trait.

`Any operation holds the AffineSymbol, and its result can be used as a symbol.` -> `The results of an operation with the ... trait can be used as symbols for the purposes for affine dialect purposes.`

Reply:
The question of why something specific to the affine dialect and analysis is in `OpDefinition.h` would arise. Ideally, traits are something that are very core to the `Operation`. OTOH, this property should ideally be an `Interface`, i.e., simply create an `AffineSymbolInterface` under `lib/Interfaces/` for this. You can then do: `isa<AffineSymbolInterface>(...)` instead of `hasTrait<...>`. However, it won't have verifiers then.

Reply:
I think think verifiers are still needed and the type of results needs to be checked.I apologize for my poor powers of expression.